src/OpenTracing/Mock/MockTracer.php

```diff
@@ -69,7 +70,11 @@ public function startSpan(string $operationName, $options = []): Span
         if (empty($options->getReferences())) {
             $spanContext = MockSpanContext::createAsRoot();
         } else {
-            $spanContext = MockSpanContext::createAsChildOf($options->getReferences()[0]);
+            $referenceContext = $options->getReferences()[0]->getContext();
+            if (!$referenceContext instanceof MockSpanContext) {
```

Comment:
I am always afraid of doing `if (!$something instanceof ...)` I usually do `if (!($something instanceof ...))`. Would you add a failing test case for this?

Reply:
Is there a reason for your fear? [The official PHP documentation](https://www.php.net/manual/en/language.operators.precedence.php) is very clear about operator precedence. :) 

But sure, I will add a test case for the exception branch.

Reply:
interesting I did not know about it, I guess my fear was because the dynamic typing.